### PR TITLE
Improve audio handling on iOS

### PR DESCRIPTION
**Pull request by davidzhao**
iOS: enabled audio playback mode without recording
iOS: Fixes audio handling with lightning headphones #687

### CODE REVIEW

File: common/darwin/Classes/FlutterRTCPeerConnection.m [hunk 1]
```diff
@@ -388,19 +390,25 @@ - (void)peerConnection:(RTCPeerConnection *)peerConnection didAddStream:(RTCMedi
...
         [videoTracks addObject:@{@"id": track.trackId, @"kind": track.kind, @"label": track.trackId, @"enabled": @(track.isEnabled), @"remote": @(YES), @"readyState": @"live"}];
     }
-    
+
     NSString *streamId = stream.streamId;
     peerConnection.remoteStreams[streamId] = stream;
-    
+
+    if (hasAudio) {
+        [AudioUtils ensureAudioSessionWithRecording:NO];
```
Comment (cloudwebrtc) [inline common/darwin/Classes/FlutterRTCPeerConnection.m:409]: When `didAddStream` was triggered, we received a `remote track`, but before that, we had created a `local track` to capture audio. Is there any wrong result here?

Comment (davidzhao) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: that case works fine, `ensureAudioSessionWithRecording` will always upgrade the audio session, and will never downgrade.

* when we received a track: sets to Playback
* then create local track: sets to PlayAndRecord
* then received another track: does nothing

Comment (cloudwebrtc) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: I think there is an easier way:

```
@interface FlutterWebRTCPlugin : NSObject<FlutterPlugin, RTCPeerConnectionDelegate>
...

-(void) ensureAudioSessionWithRecording {
  BOOL hasRecordingTrack = NO;
  for (RTCMediaStreamTrack *track in self.localTracks) {
         if([track isKindOfClass:[RTCAudioTrack class]]) {
                hasRecordingTrack = YES;
         }
   }

 BOOL hasPlaybackTrack = NO;
 for(RTCPeerConnection *pc in self.peerConnections) {
          for (RTCMediaStreamTrack *track in pc.remoteTracks) {
             if([track isKindOfClass:[RTCAudioTrack class]]) {
                    hasPlaybackTrack = YES;
             }
      }
 }
 if(hasRecordingTrack ... hasPlaybackTrack) {
     //do something here.
  }
}
```

Comment (davidzhao) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: sure, that works too!

Comment (cloudwebrtc) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: Then the `ensureAudioSessionWithRecording` method is called after the number of any remote/local track changes.

Comment (cloudwebrtc) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: @davidzhao What do you think about it?

**Review by cloudwebrtc — APPROVED** [review]

File: common/darwin/Classes/FlutterRTCPeerConnection.m (hunk 1 above)
Comment (davidzhao) [reply common/darwin/Classes/FlutterRTCPeerConnection.m:409]: sure, I can update it

**Comment by davidzhao** [comment]
@kNoAPP I just pushed a fix to a new PR #750. Please give it a try locally and let me know if that fixes things for you.

**Comment by kNoAPP** [comment]
@davidzhao thanks. I'll pull the PR and test it shortly!

**Comment by davidzhao** [comment]
@kNoAPP thanks for confirming! I'm glad this fixed it for good.

**Pull request merged**